### PR TITLE
[feature/share-extension-account-items] Expanded Single Account / Additional Items

### PR DESCRIPTION
**Pull request by hosy**
## Description
- expand the account section, if only one account is configured
- show quick access and saved searches in client location picker in share extension to get more access to save locations
- automatically expand quick access folder in share sheet

Simulator Screenshot - iPhone 15 Pro - 2023-11-16 at 18 32 07-2

## Related Issue
https://github.com/owncloud/enterprise/issues/6134

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] Added an issue with details about all relevant changes in the [**iOS documentation repository**](https://github.com/owncloud/docs-client-ios-app/issues).
- [ ] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Added changelog files for the fixed issues in folder changelog/unreleased

### CODE REVIEW

File: ownCloudAppShared/Client/Account/Controller/AccountController.swift [hunk 1]
```diff
@@ -472,6 +475,11 @@ public class AccountController: NSObject, OCDataItem, OCDataItemVersioning, Acco
 				if let quickAccessFolderDataSource = specialItemsDataSources[.quickAccessFolder] {
 					sources.append(quickAccessFolderDataSource)
 				}
+				
+				if configuration.expandQuickAccess, let accountControllerSection = accountControllerSection,
+				   let expandedItemRefs = accountControllerSection.collectionViewController?.wrap(references: [  specialItemsDataReferences[.quickAccessFolder]! ], forSection: accountControllerSection.identifier) {
+					accountControllerSection.expandedItemRefs = expandedItemRefs
+				}
```
Comment (felix-schwarz) [inline ownCloudAppShared/Client/Account/Controller/AccountController.swift:482]: Ideally `expandedItems` should not be modified or set directly, but via `.addExpanded()` and `.removeExpanded()`:

```suggestion
				if configuration.expandQuickAccess, let accountControllerSection = accountControllerSection,
				   let quickAccessItemRef = accountControllerSection.collectionViewController?.wrap(references: [  specialItemsDataReferences[.quickAccessFolder]! ], forSection: accountControllerSection.identifier).first {
					accountControllerSection.addExpanded(item: quickAccessItemRef)
				}
```

**Review by felix-schwarz — CHANGES_REQUESTED** [review]

**Comment by jesmrec** [comment]
### (1)

Menu is only expanded the first time. Follow these steps:

1. Add only one account to device
2. Open `Photos` (or any other app to share files)
3. Select some pictures to share with oC -> Picker displayed
4. `Cancel`
5. Repeat step 3. 

Current: menu not expanded:

https://github.com/owncloud/ios-app/assets/14894746/fc996048-04ca-43b6-b512-bbe126bdcbf9

Expected: menu expanded, no matter hay many times

iPhoneXR v17.2
`23b11e12`

**Comment by michaelstingl** [comment]
> Current: menu not expanded:

@felix-schwarz could you have another 👀 ?

**Comment by jesmrec** [comment]
Checked the fix, please take a look to the following video ( taken with `ecf3610a - 57b05c6`):

https://github.com/owncloud/ios-app/assets/14894746/950d978d-de62-4327-9ec2-78f7175a6d69

- First time sharing, menu is expanded but then collapsed
- Second time, `Connecting` in the status bar ant the app closes itself
- Third time, same as first.

I tested many times, and also worked fine in some of then. Not sure if there are kind of race condition or similar that causes such weird effect. If logs or any other info helps, i will catch.

Tested with iPhoneXR 17.2

**Comment by felix-schwarz** [comment]
@jesmrec I had another look and it seems those were weird timing issues, which I hope to have addressed with [b828930](https://github.com/owncloud/ios-app/pull/1296/commits/b828930ac52f77174e31d66b5a86089d45e413eb).  Please have another look.

**Comment by jesmrec** [comment]
works much better, but i still see that is not totally correct. Sometimes is not expanded

https://github.com/owncloud/ios-app/assets/14894746/57f3cfa5-c75a-4b64-9dab-8e02b112a022

`52c4fc7`
`74156a0`

**Comment by felix-schwarz** [comment]
Superseded by changes in #1320.

**Pull request closed without merging**